apollo-federation/src/composition/satisfiability/validation_state.rs

```diff
@@ -221,35 +248,232 @@ impl ValidationState {
+                                for subgraph in partitioned.keys() {
+                                    errors_by_subgraph.entry(subgraph.clone()).or_default();
+                                }
+
+                                ValidationState {
+                                    supergraph_path: new_supergraph_path,
+                                    subgraph_path_infos: SubgraphPathInfos::TopLevelMutationField {
+                                        mutation_field,
+                                        paths: partitioned,
+                                    },
+                                    selected_override_conditions: new_override_conditions,
+                                }
+                            }
+                        } else {
+                            ValidationState {
+                                supergraph_path: new_supergraph_path,
+                                subgraph_path_infos: SubgraphPathInfos::Paths(new_subgraph_paths),
+                                selected_override_conditions: new_override_conditions,
+                            }
+                        }
+                    }
+                    ValidationResult::Error { dead_ends } => {
+                        satisfiability_error(
+                            &new_supergraph_path,
+                            &subgraph_paths
+                                .iter()
+                                .map(|path| path.path.path.as_ref())
+                                .collect::<Vec<_>>(),
+                            &dead_ends,
+                            errors,
+                        )?;
+                        return Ok(None);
+                    }
+                }
+            }
+            SubgraphPathInfos::TopLevelMutationField {
+                mutation_field,
+                paths: partitioned_paths,
+            } => {
+                let mut new_partitioned: IndexMap<Arc<str>, Vec<SubgraphPathInfo>> =
+                    Default::default();
+
+                for (subgraph, subgraph_path_infos) in partitioned_paths.iter_mut() {
+                    let errors_for_subgraph = satisfiability_errors_by_mutation_field_and_subgraph
+                        .get_mut(mutation_field)
+                        .and_then(|m| m.get_mut(subgraph));
+
+                    let result = Self::validate_transition_for_subgraph_paths(
+                        supergraph_schema,
+                        subgraph_path_infos,
+                        &new_override_conditions,
+                        &edge_weight.transition,
+                        target_type,
+                        matching_contexts,
+                        condition_resolver,
+                    )?;
+
+                    match result {
+                        ValidationResult::Success { new_subgraph_paths }
+                            if new_subgraph_paths.is_empty() =>
+                        {
```

Comment:
Another [comment](https://github.com/apollographql/federation/pull/3305/files#diff-5c7c66b4e2f5f913f1c12534c861af5a519f666e0e11af4756934d3b49aa52c1R715) to copy here.